payments-ui-core/src/main/java/com/stripe/android/ui/core/FormUI.kt

```diff
@@ -74,82 +75,161 @@ fun FormUI(
-            modifier = Modifier.padding(4.dp, 8.dp, 4.dp, 4.dp),
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                start = 4.dp,
+                end = 4.dp,
+                top = 8.dp,
+                bottom = 4.dp,
+            ),
         )
         is AuBecsDebitMandateTextElement -> AuBecsDebitMandateElementUI(
             element = element,
-            modifier = Modifier.padding(vertical = 8.dp),
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                vertical = 8.dp,
+            ),
         )
         is AffirmHeaderElement -> AffirmElementUI(
-            modifier = Modifier.padding(vertical = 8.dp),
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                vertical = 8.dp,
+            ),
         )
         is MandateTextElement -> MandateTextUI(
             element = element,
-            modifier = Modifier.padding(top = element.topPadding, bottom = 8.dp)
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                top = element.topPadding,
+                bottom = 8.dp,
+            ),
         )
         is CardDetailsSectionElement -> CardDetailsSectionElementUI(
-            enabled,
-            element.controller,
-            hiddenIdentifiers,
-            lastTextFieldIdentifier
+            enabled = enabled,
+            controller = element.controller,
+            hiddenIdentifiers = hiddenIdentifiers,
+            lastTextFieldIdentifier = lastTextFieldIdentifier,
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                vertical = 8.dp,
+            ),
         )
         is BsbElement -> BsbElementUI(
             enabled = enabled,
             element = element,
             lastTextFieldIdentifier = lastTextFieldIdentifier,
-            modifier = Modifier.padding(vertical = 8.dp),
+            modifier = Modifier.addFormVerticalPadding(
+                size = size,
+                index = index,
+                vertical = 8.dp,
+            ),
         )
         is OTPElement -> OTPElementUI(enabled, element)
         is RenderableFormElement -> element.ComposeUI(enabled)
-        is EmptyFormElement -> {}
     }
 }
+
+private fun Modifier.addFormVerticalPadding(
+    index: Int,
+    size: Int,
+    vertical: Dp,
+) = addFormVerticalPadding(
+    index = index,
+    size = size,
+    top = vertical,
+    bottom = vertical
+)
+
+private fun Modifier.addFormVerticalPadding(
+    index: Int,
+    size: Int,
+    top: Dp,
+    bottom: Dp,
+    start: Dp = 0.dp,
+    end: Dp = 0.dp,
+) = when {
+    index == 0 && size == 1 -> this.padding(top = 0.dp, bottom = 0.dp, start = start, end = end)
```

Comment:
I think you can remove the `this.` from these.

Reply:
I actually got a Jetpack Compose warning about not using `this` in the `Modifier` context which is why I added it.